core/templates/pages/exploration-player-page/learner-experience/tutor-card.directive.spec.ts

```diff
@@ -0,0 +1,420 @@
+      writtenTranslationsObjectFactory.createEmpty(),
+      'content', audioTranslationLanguageService);
+    spyOnProperty(explorationPlayerStateService, 'onOppiaFeedbackAvailable')
+      .and.returnValue(oppiaFeedbackEmitter);
+    spyOnProperty(playerPositionService, 'onActiveCardChanged').and.returnValue(
+      activeCardChangedEmitter);
+    spyOnProperty(playerPositionService, 'onNewCardAvailable').and.returnValue(
+      newCardAvailableEmitter);
+    spyOn(deviceInfoService, 'isMobileDevice').and.returnValue(true);
+    spyOn(currentInteractionService, 'registerPresubmitHook')
+      .and.callFake((cb) => {
+        cb();
+      });
+
+    directive = $injector.get('tutorCardDirective')[0];
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $scope: $scope,
+      $rootScope: $scope,
+      $anchorScroll: $anchorScroll,
+      LearnerAnswerInfoService: learnerAnswerInfoService
+    });
+
+    $scope.getDisplayedCard = function() {
+      return sampleCard;
+    };
+    $scope.getDisplayedCard().markAsCompleted();
+  }));
+
+  beforeEach(() => {
+    testSubscriptions = new Subscription();
+    testSubscriptions.add(
+      explorationPlayerStateService.onOppiaFeedbackAvailable.subscribe(
+        feedbackSpy));
+  });
+
+  afterEach(() => {
+    testSubscriptions.unsubscribe();
+    ctrl.$onDestroy();
+  });
+
+  describe('on initialization ', function() {
+    it('should set profile picture from data url when ' +
+      'user is not in editor preview mode', fakeAsync(function() {
+      spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+      spyOn(userService, 'getProfileImageDataUrlAsync')
+        .and.resolveTo('imageUrl');
+
+      ctrl.$onInit();
+      oppiaFeedbackEmitter.emit();
+      activeCardChangedEmitter.emit();
+      newCardAvailableEmitter.emit();
+      tick();
+
+      expect($scope.profilePicture).toBe('imageUrl');
+    }));
+
+    it('should set default profile picture when ' +
+      'user is in editor preview mode', fakeAsync(function() {
+      spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(true);
+      spyOn(urlInterpolationService, 'getStaticImageUrl')
+        .and.returnValue('staticImage');
+
+      ctrl.$onInit();
+      oppiaFeedbackEmitter.emit();
+      activeCardChangedEmitter.emit();
+      newCardAvailableEmitter.emit();
+      tick();
+
+      expect($scope.profilePicture).toBe('staticImage');
+    }));
+
+    it('should check whether the audio bar is expanded on ' +
+      'mobile device', fakeAsync(function() {
+      spyOn(contextService, 'isInExplorationEditorPage')
+        .and.returnValue(false);
+      spyOn(userService, 'getProfileImageDataUrlAsync')
+        .and.resolveTo('imageUrl');
+
+      ctrl.$onInit();
+      tick();
+      let result = $scope.isAudioBarExpandedOnMobileDevice();
+
+      expect(result).toBe(false);
+    }));
+  });
+
+  it('should check whether the display card ' +
```

Comment:
For this and rest of the tests add a space between:

1. setup
2. pre-checks
3. action
4. post-checks

Reply:
done